src/lib/serviceRoutines/statisticsTreat.cpp

```diff
@@ -143,261 +148,272 @@ std::string statisticsTreat
+    out += TAG_ADD_COUNTER("individualContextEntityAttributeWithTypeAndId", noOfIndividualContextEntityAttributeWithTypeAndId);
   }
 
   if (noOfAttributeValueInstanceWithTypeAndId != -1)
   {
-    out += TAG_ADD("attributeValueInstanceWithTypeAndId", noOfAttributeValueInstanceWithTypeAndId);
+    out += TAG_ADD_COUNTER("attributeValueInstanceWithTypeAndId", noOfAttributeValueInstanceWithTypeAndId);
   }
 
   if (noOfContextEntitiesByEntityIdAndType != -1)
   {
-    out += TAG_ADD("contextEntitiesByEntityIdAndType", noOfContextEntitiesByEntityIdAndType);
+    out += TAG_ADD_COUNTER("contextEntitiesByEntityIdAndType", noOfContextEntitiesByEntityIdAndType);
   }
 
   if (noOfEntityByIdAttributeByNameIdAndType != -1)
   {
-    out += TAG_ADD("entityByIdAttributeByNameIdAndType", noOfEntityByIdAttributeByNameIdAndType);
+    out += TAG_ADD_COUNTER("entityByIdAttributeByNameIdAndType", noOfEntityByIdAttributeByNameIdAndType);
   }
 
   if (noOfLogRequests != -1)
   {
-    out += TAG_ADD("logRequests", noOfLogRequests);
+    out += TAG_ADD_COUNTER("logRequests", noOfLogRequests);
   }
 
   if (noOfVersionRequests != -1)
   {
-    out += TAG_ADD("versionRequests", noOfVersionRequests);
+    out += TAG_ADD_COUNTER("versionRequests", noOfVersionRequests);
   }
 
   if (noOfExitRequests != -1)
   {
-    out += TAG_ADD("exitRequests", noOfExitRequests);
+    out += TAG_ADD_COUNTER("exitRequests", noOfExitRequests);
   }
 
   if (noOfLeakRequests != -1)
   {
-    out += TAG_ADD("leakRequests", noOfLeakRequests);
+    out += TAG_ADD_COUNTER("leakRequests", noOfLeakRequests);
   }
 
   if (noOfStatisticsRequests != -1)
   {
-    out += TAG_ADD("statisticsRequests", noOfStatisticsRequests);
+    out += TAG_ADD_COUNTER("statisticsRequests", noOfStatisticsRequests);
   }
 
   if (noOfInvalidRequests != -1)
   {
-    out += TAG_ADD("invalidRequests", noOfInvalidRequests);
+    out += TAG_ADD_COUNTER("invalidRequests", noOfInvalidRequests);
   }
 
   if (noOfRegisterResponses != -1)
   {
-    out += TAG_ADD("registerResponses", noOfRegisterResponses);
+    out += TAG_ADD_COUNTER("registerResponses", noOfRegisterResponses);
   }
 
 
   if (noOfRegistrationErrors != -1)
   {
-    out += TAG_ADD("registrationErrors", noOfRegistrationErrors);
+    out += TAG_ADD_COUNTER("registrationErrors", noOfRegistrationErrors);
   }
 
   if (noOfRegistrationUpdateErrors != -1)
   {
-    out += TAG_ADD("registrationUpdateErrors", noOfRegistrationUpdateErrors);
+    out += TAG_ADD_COUNTER("registrationUpdateErrors", noOfRegistrationUpdateErrors);
   }
 
   if (noOfDiscoveryErrors != -1)
   {
-    out += TAG_ADD("discoveryErrors", noOfDiscoveryErrors);
+    out += TAG_ADD_COUNTER("discoveryErrors", noOfDiscoveryErrors);
+  }
+
+  if (semTimeStatistics)
+  {
+    char requestSemaphoreWaitingTime[64];
+    semTimeReqGet(requestSemaphoreWaitingTime, sizeof(requestSemaphoreWaitingTime));
+    out += TAG_ADD_STRING("requestSemaphoreWaitingTime", requestSemaphoreWaitingTime);
```

Comment:
Why in this case we use TAG_ADD_STRING? requestSemaphoreWaitingTime is not a counter? I'm missing some subtle detail...